src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Elasticsearch/RequestWrapper.cs

```diff
@@ -91,8 +78,10 @@ void InvokeTryProcessResponse(Task responseTask)
                         onSuccess: response =>
                         {
                             var uri = GetUriFromResponse(response);
-                            // TODO: need to figure out how to plumb things so that we can set the uri on the segment after it has already been created.
-                            // See comments above about extending the experimental API
+
+                            var data = segment.GetExperimentalApi().SegmentData as IDatastoreSegmentData;
```

Comment:
You only need to call `GetExperimentalApi` once if you save the reference to it.